### ISSUE TITLE
publish tries to fetch special remote

### ISSUE THREAD

**Issue by bpoldrack** [datalad/datalad#4151]
With recent change in `publish`, it tries to fetch a special remote (and fails, of course stating that it's not a git repository). Discovered here: https://travis-ci.org/datalad/datalad/jobs/650395979#L1171

Culprit seems to be here: https://github.com/datalad/datalad/blob/master/datalad/distribution/publish.py#L295
This line tests `remote` to be an annex, instead of `r`.

**Comment by kyleam** [comment]
Your name swap in be4a512 looks obviously correct, but "with recent change in publish" made me wonder what exactly changed.  The lines you point to haven't changed since 2017.  It looks like what is going on is that, as of the recently merged 5b506396fc (RF: GitRepo.fetch/push/pull() without GitPython), fetching a remote without a URL will raise an exception rather than return an empty list.  Here's the updated test from that commit:

```diff
diff --git a/datalad/support/tests/test_gitrepo.py b/datalad/support/tests/test_gitrepo.py
index becb76a477..0c26f008e0 100644
--- a/datalad/support/tests/test_gitrepo.py
+++ b/datalad/support/tests/test_gitrepo.py
@@ -519,9 +519,7 @@ def test_GitRepo_fetch(test_path, orig_path, clone_path):
     origin.config.unset('remote.not-available.url', where='local')
 
     # fetch without provided URL
-    fetched = origin.fetch('not-available')
-    # nothing was done, nothing returned:
-    eq_([], fetched)
+    assert_raises(CommandError, origin.fetch, 'not-available')
```

**Comment by bpoldrack** [comment]
True. Thx for digging what's actually changed.
